core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -449,7 +450,11 @@ public String getMaxValue() {
 
         @Override
         protected java.lang.Long valueOf(String value) {
-            return dateFormatter.parser().parseMillis(value);
+            try {
+                return dateFormatter.parser().parseMillis(value);
+            } catch (Exception e) {
```

Comment:
+1 to *not* build in another hard coded fall back (we don't know what a number means - might be seconds). I do wonder what goes wrong here. The default date field formatter should fall back to mills:  `strict_date_optional_time||epoch_millis` .